src/libraries/Native/Unix/CMakeLists.txt

```diff
@@ -44,6 +44,11 @@ if (COMPILER_SUPPORTS_W_IMPLICIT_FALLTHROUGH)
   add_compile_options(-Wimplicit-fallthrough)
 endif()
 
+check_c_compiler_flag(-Wno-strict-prototypes HAVE_WNO_STRICT_PROTOTYPES)
+if (HAVE_WNO_STRICT_PROTOTYPES)
+    add_compile_options(-Wno-strict-prototypes)
+endif()
+
```

Comment:
I can test against this, but if memory serves building clang-15 fails without this.

Reply:
Warning-as-errors are disabled in release branches so that we do not have backport C++ compiler warning tweaks like this one.

https://github.com/dotnet/runtime/blob/release/6.0/eng/native/configureplatform.cmake#L5
https://github.com/dotnet/runtime/blob/release/6.0/eng/native/configurecompiler.cmake#L324-L327

If this does not work as expected, I would like to understand the reason.

Reply:
@ayakael, just curious, was the change here intended for this PR? The description mentions it's a backport of https://github.com/dotnet/runtime/pull/73065 and https://github.com/dotnet/runtime/pull/73905, but those PRs don't have this change.

Reply:
It was added in .NET 6.0.110 as without it our build failed on Alpine with clang-15. I could retest it.

Reply:
Through retesting it is indeed not necessary. It was at some point, but fortunately not anymore. Pushed new version without this change.